core/services/ocr2/plugins/ccip/config/config.go

```diff
@@ -42,8 +43,134 @@ func (c CommitPluginConfig) Encode() ([]byte, error) {
 
 // DynamicPriceGetterConfig specifies which configuration to use for getting the price of tokens (map keys).
 type DynamicPriceGetterConfig struct {
+	// Deprecated: use TokenPrices instead.
 	AggregatorPrices map[common.Address]AggregatorPriceConfig `json:"aggregatorPrices"`
-	StaticPrices     map[common.Address]StaticPriceConfig     `json:"staticPrices"`
+	// Deprecated: use TokenPrices instead.
+	StaticPrices map[common.Address]StaticPriceConfig `json:"staticPrices"`
+
+	TokenPrices []TokenPriceConfig `json:"tokenPrices"`
+}
+
+// IsDeprecated returns true if the config uses the deprecated fields.
+func (c *DynamicPriceGetterConfig) IsDeprecated() bool {
+	return len(c.AggregatorPrices) > 0 || len(c.StaticPrices) > 0
+}
+
+// TokenPriceConfig specifies the configuration for a token price.
+type TokenPriceConfig struct {
+	// TokenAddress is the address of the token on the chain that is deployed on.
+	TokenAddress common.Address `json:"tokenAddress"`
```

Comment:
Any reason we don't use `cciptypes.Address` here? I notice we do the back/forth conversion in a bunch of places. Is it for consistency with the `AggregatorPriceConfig` type?

I guess you can even make use of `TokenID` here if you can use the `cciptypes.Address` type, 

```golang
type TokenPriceConfig struct {
  TokenID ccipcommon.TokenID
  AggregatorConfig ...
  ...
}
```